### PR TITLE
Switch to sbt-microsites #438

### PR DESCRIPTION
**Pull request by jonas**
Preview here: https://jonas.github.io/circe/

Let me know if navigation is OK. It retains the "menu" of the current site and adds the contribution page.

List of things to fix:
 - [x] Add PNGs with circe icon and jumbotron pattern
 - [x] Better tag line than "Yet another JSON lib ..."?
 - [x] Experiment with custom color scheme?
 - [x] Fix links to Scaladoc on index page to work for Scala 2.12

A small contribution as a thank you for making circe! I especially like how the enum codecs in 0.6.0 made integration with Sangria and Scala.js so much simpler.

### CODE REVIEW

**Comment by travisbrown** [comment]
Oh wow, this is so great, @jonas! Navigation looks good to me, and the to-do list looks perfect.

**Comment by jonas** [comment]
I time boxed the tasks and didn't find a good alternative color scheme so I ended up creating an version of the icon with alternative colors. 

It conflicts with some of the changes in #467 so a rebase is probably needed if that gets merged.

File: docs/src/main/resources/microsite/includes/references.md [hunk 1]
```diff
@@ -3,38 +3,38 @@
...
-[circe-generic]: https://travisbrown.github.io/circe/api/#io.circe.generic.auto$
-[circe-jackson]: https://travisbrown.github.io/circe/api/#io.circe.jackson.package
-[circe-jawn]: https://travisbrown.github.io/circe/api/#io.circe.jawn.package
-[circe-parser]: https://travisbrown.github.io/circe/api/#io.circe.parser.package
+[circe-251]: https://github.com/circe/circe/issues/251
+[circe-262]: https://github.com/circe/circe/issues/262
+[circe-generic]: https://circe.github.io/circe/api/io/circe/generic/auto$.html
+[circe-jackson]: https://circe.github.io/circe/api/io/circe/jackson/index.html
+[circe-jawn]: https://circe.github.io/circe/api/io/circe/jawn/index.html
+[circe-parser]: https://circe.github.io/circe/api/io/circe/parser/index.html
```
Comment (jonas) [inline docs/src/main/resources/microsite/includes/references.md]: For some reason the circe-parser package doesn't show up in the scaladoc. Didn't investigate further.

**Comment by travisbrown** [comment]
Looks good to me, @jonas—we can refine the color scheme later. I'll go ahead and merge this when it's green and then rebase #467.

**Comment by jonas** [comment]
👍

File: docs/src/main/resources/microsite/includes/references.md [hunk 2]
```diff
@@ -3,38 +3,38 @@
 [akka-http-json]: https://github.com/hseeberger/akka-http-json
 [akka-stream-json]: https://github.com/knutwalker/akka-stream-json
 [apache]: http://www.apache.org/licenses/LICENSE-2.0
-[api]: https://travisbrown.github.io/circe/api/#io.circe.package
+[api]: https://circe.github.io/circe/api/io/circe/package.html
```
Comment (jonas) [inline docs/src/main/resources/microsite/includes/references.md]: Oops sorry I missed this one, should be index.html instead of package.html.

**Pull request merged**